Octokit/Models/Request/Enterprise/UpdateMaintenanceRequest.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+
+namespace Octokit
+{
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public class UpdateMaintenanceRequest
+    {
+        /// <summary>
+        /// Maintenance request with default values (results in Maintenance mode being turned off immediately)
+        /// </summary>
+        public UpdateMaintenanceRequest()
+        { }
+
+        /// <summary>
+        /// Maintenance request specifying whether to enable/disable maintenance and when (only applicable when enabling maintenance)
+        /// </summary>
+        /// <param name="enabled">true to enable, false to disable</param>
+        /// <param name="when">A <see cref="MaintenanceDate"/> object specifying when maintenance mode will be enabled.
+        /// Use static helper methods
+        ///   <see cref="MaintenanceDate.Now()"/>
+        ///   <see cref="MaintenanceDate.FromDateTimeOffset(DateTimeOffset)"/> and
+        ///   <see cref="MaintenanceDate.FromChronicValue(string)"/></param>
+        /// <remarks>If enabled is false, the when parameter is ignored and maintenance is turned off immediately</remarks>
+        public UpdateMaintenanceRequest(bool enabled, MaintenanceDate when)
+        {
+            Enabled = enabled;
+            When = when;
+        }
+
+        public bool Enabled { get; protected set; }
+        
+        public MaintenanceDate When { get; protected set; }
+
+        [SuppressMessage("Microsoft.Globalization", "CA1304:SpecifyCultureInfo", MessageId = "System.String.ToLower")]
+        public string AsJsonString()
+        {
+            var json = string.Format(CultureInfo.InvariantCulture,
+                "\"enabled\":{0}",
+                Enabled.ToString().ToLower());
+
+            if (Enabled)
+            {
+                json = string.Concat(json, string.Format(CultureInfo.InvariantCulture,
+                ", \"when\":\"{0}\"",
+                When.Value));
+            }
+
+            json = string.Concat("{", json, "}");
+
+            return json;
+        }
+
+        public string AsNamedFormEncodingString()
+        {
+            var value = string.Format(CultureInfo.InvariantCulture, "maintenance={0}", this.AsJsonString());
```

Comment:
i was thinking perhaps like how we have the `RequestParameters` class that has the `ToParametersDictionary()` that we could maybe extend this (or add a new similar class) that would provide a `ToFormUrlEncoded()` method on it, which would provide a way to have any request class be able to have it's key/value pairs formatted up to suit the required style here (eg  key=value&key2=value2)